File: 0258-add-digits/0258-add-digits.java
class Solution {
    public int addDigits(int num) {
        String numString=String.valueOf(Math.abs(num));
        int numLength=numString.length();
        while(numLength>1){
            int sum=0;
            for(int i=0;i<numLength;i++){
            int remainder=num%10;
            sum+=remainder;
            num=num/10;
            }
            num=sum;
            numString=String.valueOf(Math.abs(num));
            numLength=numString.length();
        }
        return num;
    }
}

File: 0258-add-digits/0258-add-digits.py
class Solution(object):
    def addDigits(self, num):
        '''O(logn) approach-
        sum=0
        if num==0:
            return 0
        while num>0:
            sum+=num%10
            num//=10
        if sum>9:
            return self.addDigits(sum)
        return sum'''

        #O(1) approach -
        if num<=9:
            return num
        if num%9==0:
            return 9
        return num%9
        
        

        